packages/grid/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -230,12 +230,18 @@ function extractColumnWidths(
       if (header) {
         const title = header.querySelector(`.${gridClasses.columnHeaderTitle}`);
         const content = header.querySelector(`.${gridClasses.columnHeaderTitleContainerContent}`)!;
+        const iconContainer = header.querySelector(`.${gridClasses.iconButtonContainer}`);
+        const menuContainer = header.querySelector(`.${gridClasses.menuIcon}`);
         const element = title ?? content;
 
         const style = window.getComputedStyle(header, null);
         const paddingWidth = parseInt(style.paddingLeft, 10) + parseInt(style.paddingRight, 10);
         const contentWidth = element.scrollWidth + 1;
-        const width = paddingWidth + contentWidth;
+        const width =
+          contentWidth +
+          paddingWidth +
+          (iconContainer?.clientWidth ?? 0) +
+          (menuContainer?.clientWidth ?? 0);
```

Comment:
I think we'll have to differentiate the logic for desktop & mobile. We only keep the icons visible at all times on mobile, and we don't want to pay the width cost on desktop.

Reply:
@romgrk Apologies for the delay in resolving this.

The logic actually takes that into account already. Here is a [loom video](https://www.loom.com/share/ffd6b54630fa4f4682d2b1ca104c9267) to confirm